web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/CitizenCertificateRecipientConverterImpl.java

```diff
@@ -0,0 +1,44 @@
+package se.inera.intyg.intygstjanst.web.service.impl;
+
+import org.springframework.stereotype.Service;
+import se.inera.intyg.intygstjanst.web.service.CitizenCertificateRecipientConverter;
+import se.inera.intyg.intygstjanst.web.service.bean.CertificateRecipientType;
+import se.inera.intyg.intygstjanst.web.service.dto.citizen.CitizenCertificateRecipientDTO;
+import se.inera.intyg.intygstjanst.web.service.repo.RecipientRepo;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class CitizenCertificateRecipientConverterImpl implements CitizenCertificateRecipientConverter {
+
+    private final RecipientRepo recipientRepo;
+
+    public CitizenCertificateRecipientConverterImpl(RecipientRepo recipientRepo) {
+        this.recipientRepo = recipientRepo;
+    }
+
+    @Override
+    public Optional<CitizenCertificateRecipientDTO> convert(String certificateType, LocalDateTime sent) {
+        final var recipients = recipientRepo
+                .listRecipients()
+                .stream()
+                .filter(
```

Comment:
See if you can just return the "HUVUDMOTTAGARE" recipient instead of a list. The "recipients.get(0)" down below is a bit concerning... It is dependent on that the HUVUDMOTTAGARE is the first element... but should the code really trust that.

Reply:
In the filter the list is minimized to only HUVUDMOTTAGARE so it doesn't depend on order. Do you mean I should add a new function in RecipientRepo?

Reply:
Good suggestion, I updated this